lib/kcerts/ssh.go

```diff
@@ -103,6 +106,57 @@ func (a SSHAgent) Valid() bool {
 	return err == nil
 }
 
+func (a *SSHAgent) GetStandardSocketPath() (string, error) {
+	path, err := GetConfigDir("enkit")
+	if err != nil {
+		return "", err
+	}
+
+	// Securely make sure path exists and is permission 0700
+	if err := os.MkdirAll(path, 0700); err != nil && !os.IsExist(err) {
+		return "", err
+	}
+	if err := os.Chmod(path, 0700); err != nil {
+		return "", err
+	}
+
+	socket := filepath.Join(path, "agent")
+
+	return socket, nil
+}
+
+func (a *SSHAgent) UseStandardPaths() error {
+	socket, err := a.GetStandardSocketPath()
+	if err != nil {
+		return err
+	}
+
+	if a.Socket == socket {
+		// no standardization needed.
+		return nil
+	}
+
+	// Securely rename socket:
```

Comment:
there's a race condition, and as per comment above, I think we don't need to be as paranoid to write in the home directory of the user. If others can write there/predict a path, there are bigger problems.

I would either:
1) Just create the symlink (remove old one, if necessary)
2) Or rename in place (so it's atomic).

Something like:
```
tempname := fmt.Sprintf("%s.tmp%016x", rand.New(srand.Source).Uint64())
defer os.Remove(tempname)
if err := os.Symlink(a.Socket, tempname); err != nil {
...
}
if err := os.Rename(tempname, socket); err != nil {
...
}
```

Reply:
Makes sense.  Done.